### PR TITLE
Task/add time instant metadata

### LINKED ISSUE #26: Add TimeInstant metadata
Add a TimeInstant metadata to every attribute, with the timestamp of the modification, as shown in the following example:
```
  {
            "id":"<entity-name>",
            "type" : "<entity-type>",
            "isPattern" : "false",
            "attributes" : [
                           {
             "name":"_TTcurrent_conf",
             "type":"string",
             "value":"44",
             "metadatas":[
                {
                 "name":"TimeInstant",
                 "type":"ISO8601",
                 "value":"2015-03-19T14:39:14.924961Z"
                }
                ]
            },
               {
             "name":"TimeInstant",
             "type":"ISO8601",
              "value":"2015-03-19T14:39:14.924961Z"
            }]
    }
```

### PR DESCRIPTION
**Pull request by dmoranj**
Fixes #26 

Adds the option of adding the timestamp in every attribute of every module.

### CODE REVIEW

File: lib/services/thinkingParser.js [hunk 1]
```diff
@@ -34,17 +35,19 @@ var errors = require('../errors'),
...
 
-function parseHumidity(fields, result, callback) {
+function parseHumidity(fields, result, timestamp, callback) {
     if (fields.length >= 5) {
         result.attributes.push({
             name: 'humidity',
             value: fields[3],
-            type: 'float'
+            type: 'float',
+            metadatas: timestamp
```
Comment (gtorodelvalle) [inline lib/services/thinkingParser.js]: It is kind of a minor issue so I leave it a your discretion, but "metadatas" in plural felt weird to me when I read it... I would say there is no a plural form for "metadata"... :) The point is that there are a lot of places when you should change this if you consider it appropriate... :(

**Comment by gtorodelvalle** [comment]
LGTM although I left a minor comment... :) Once you make a decision about it or push a fix, I'll land the PR... :)

File: lib/services/thinkingParser.js (hunk 1 above)
Comment (dmoranj) [inline lib/services/thinkingParser.js]: The problem here is that this attribute was not defined by me but in the
NGSI specification. So Not To Change.

Comment (gtorodelvalle) [inline lib/services/thinkingParser.js]: Perfect! ;) BTW, I just saw a couple of missing final blank lines in a couple of files... :) I'll comment on the precise line... ;) Thanks!

File: test/unit/contextRequests/updateContextPlainTimestampExample.json [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        {
+          "name": "sleeptime",
+          "type": "string",
+          "value": "300"
+        }
+      ]
+    }
+  ],
+  "updateAction": "APPEND"
+}
```
Comment (gtorodelvalle) [inline test/unit/contextRequests/updateContextPlainTimestampExample.json]: Final blank line appreciated... :)

File: test/unit/contextResponses/updateContextPlainTimestampExampleSuccess.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        "isPattern": "false",
+        "type": "ThinkingThing"
+      },
+      "statusCode": {
+        "code": "200",
+        "reasonPhrase": "OK"
+      }
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/contextResponses/updateContextPlainTimestampExampleSuccess.json]: Final blank line appreciated... :)

File: test/unit/contextResponses/updateContextTemperatureTimestampSuccess.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        "isPattern": "false",
+        "type": "ThinkingThing"
+      },
+      "statusCode": {
+        "code": "200",
+        "reasonPhrase": "OK"
+      }
+    }
+  ]
+}
```
Comment (gtorodelvalle) [inline test/unit/contextResponses/updateContextTemperatureTimestampSuccess.json]: Final blank line appreciated... :)

File: test/unit/contextRequests/updateContextPlainTimestampExample.json (hunk 2 above)
Comment (dmoranj) [inline test/unit/contextRequests/updateContextPlainTimestampExample.json]: Fixed in 39e7a15

File: test/unit/contextResponses/updateContextPlainTimestampExampleSuccess.json (hunk 3 above)
Comment (dmoranj) [inline test/unit/contextResponses/updateContextPlainTimestampExampleSuccess.json]: Fixed in 39e7a15

File: test/unit/contextResponses/updateContextTemperatureTimestampSuccess.json (hunk 4 above)
Comment (dmoranj) [inline test/unit/contextResponses/updateContextTemperatureTimestampSuccess.json]: Fixed in 39e7a15

**Comment by gtorodelvalle** [comment]
Nice!!! Thanks @dmoranj ;) Landing... :)

**Pull request merged**